config/import-control.xml

```diff
@@ -162,5 +167,8 @@
     <allow class="java.lang.reflect.Field"/>
     <allow class="java.nio.charset.StandardCharsets" local-only="true"/>
     <allow pkg="com.puppycrawl.tools.checkstyle.utils" local-only="true"/>
+    <allow class="com.puppycrawl.tools.checkstyle.TreeWalkerAuditEvent"/>
+    <allow class="com.puppycrawl.tools.checkstyle.TreeWalkerFilter"/>
+    <allow class="com.puppycrawl.tools.checkstyle.filters.SuppressionXpathFilter"/>
```

Comment:
This is not good.
xpath implementation is general and do not know where and how it is used.

All this imports/dependencies are are not good to have on this level. Please move host classes to package where Main is located, as all of them are part of CLI, not related to core of xpath.
your cycle dependencies (`mvn clean site -Pno-validations && firefox target/site/dsm/index.html`):
![image](https://user-images.githubusercontent.com/812984/41500200-516c938e-7142-11e8-805c-dfed1d1f24e9.png)
vs
current https://checkstyle.org/dsm/index.html

just FYI: CLI functionality is growing, I am not sure we need special package for it right now, but eventually we will do this.

Reply:
done